### ISSUE TITLE
PyTorch `predict_proba` does not match original SGDClassifier model

### ISSUE THREAD

**Issue by EdgarMCR** [microsoft/hummingbird#413]
I have been trialing using hummingbird to convert a [SGDClassifier ](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) with  modified_huber loss to a PyTorch model. 

The predictions of the converted model match the original (and it was amazingly easy to convert, really awesome). 

However, calling `predict_proba` the output does not match. Its a factor of 10 smaller and loosely correlated with the originals model confidence. 
![plot](https://user-images.githubusercontent.com/12551620/103542070-497d3200-4e94-11eb-92b2-405bb4fac328.png)

**Is it possible to replicate both the classification as well as the confidence in the PyTorch model?**

As a minimal example, take this code:
```
import numpy as np
from sklearn.linear_model import SGDClassifier
from hummingbird.ml import convert

X = np.array([[-0.5, -1], [-1, -1], [-0.1, -0.1], [0.1, -0.2],
              [0.5, 1], [1, 1], [0.1, 0.1], [-0.1, 0.2]])
Y = np.array([1, 1, 1, 1, 2, 2, 2, 2])

clf = SGDClassifier(loss='modified_huber', max_iter=1000, tol=1e-3)
clf.fit(X, Y)

# Use Hummingbird to convert the model to PyTorch
model = convert(clf, 'pytorch')

inputs = [[-1, -1], [1, 1], [-0.2, 0.1], [0.2, -0.1]]
print('Original model')
print("clf.predict(inputs) = {}".format(clf.predict(inputs)))
print("clf.predict_proba(inputs) = \n{}".format(clf.predict_proba(inputs)))

print('Converted pytorch model')
print("model.predict(inputs) = {}".format(model.predict(inputs)))
print("model.predict_proba(inputs) = \n{}".format(model.predict_proba(inputs)))

```

The confidence outputed for the converted model is different every time but can be:
```
Original model
clf.predict(inputs) = [1 2 2 1]
clf.predict_proba(inputs) = 
[[1. 0.]
 [0. 1.]
 [0. 1.]
 [1. 0.]]
Converted pytorch model
model.predict(inputs) = [1 2 2 1]
model.predict_proba(inputs) = 
[[1.         0.        ]
 [0.         1.        ]
 [0.         1.        ]
 [0.7743539  0.22564608]]
```

**Comment by interesaaat** [comment]
Thanks for reporting this. We haven't tried with this loss so maybe something funky is happening (for SGD I believe we use float and not double, so maybe we are loosing something because of rounding errors, but 10x looks like a lot, so maybe something else is happening)

**Comment by interesaaat** [comment]
I just figured that we don't support `modified_huber` loss. I am working on adding it.

**Comment by EdgarMCR** [comment]
Wow, I am blown away by your speed! Thank you very much
